cli/src/store/yaml_store.rs

```diff
@@ -0,0 +1,77 @@
+// Copyright 2018-2020 Cargill Incorporated
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::fs::{self, File, OpenOptions};
+use std::path::PathBuf;
+
+use serde::{de::DeserializeOwned, Serialize};
+
+use super::YamlBackedStoreError;
+
+const DEFAULTS_PATH: &str = ".splinter";
+
+pub(super) trait YamlBackedStore<T: Serialize + DeserializeOwned> {
```

Comment:
This abstraction seems unnecessary. Just using serde directly by the file-backed stores would be a lot less code. The stores basically just need a load() and save() methods which should be a subset of this code to accomplish.

Reply:
Yeah, I was also a little unsure about this. The idea was to avoid code duplication, as any store yaml backed store (in this PR both FileBackedNodeStore and FileBackedDefaultValueStore) would not have to implement the same logic again. And I wanted to avoid creating an `utils` like file with some loose methods that both stores use. 

As to the point that `the stores basically just need a load() and save() `, its easy to refactor the code to look more like that. I think the stores just directly use the `read_data_from_file` (like load()) and `write_to_file`(like save()). The other methods are working more as helper methods.